cscs-checks/tools/profiling_and_debugging/notool.py

```diff
@@ -23,66 +21,79 @@ def __init__(self, lang):
             'PrgEnv-pgi': ['-O2', '-g', '-mp']
         }
         self.sourcesdir = os.path.join('src', lang)
-        self.executable = './jacobi'
         self.build_system = 'Make'
+        self.executable = './jacobi'
         # NOTE: Restrict concurrency to allow creation of Fortran modules
         if lang == 'F90':
             self.build_system.max_concurrency = 1
 
         self.num_tasks = 3
         self.num_tasks_per_node = 3
         self.num_cpus_per_task = 4
-        self.num_iterations = 200
+        self.num_tasks_per_core = 1
+        self.use_multithreading = False
+        self.num_iterations = 100
         self.variables = {
             'OMP_NUM_THREADS': str(self.num_cpus_per_task),
             'ITERATIONS': str(self.num_iterations),
             'OMP_PROC_BIND': 'true',
             'CRAYPE_LINK_TYPE': 'dynamic'
         }
+        # OpenMP support varies between compilers:
         self.openmp_versions = ScopedDict({
             'PrgEnv-cray': {'version': 201511},
             'PrgEnv-gnu': {'version': 201511},
             'PrgEnv-intel': {'version': 201611},
             'PrgEnv-pgi': {'version': 201307},
         })
-        # a scopedict is better than this:
-        # if (self.language == 'C++' and
+        self.lang = lang
+        # The scopedict (above) is better than this:
+        # if (self.lang == 'C++' and
         #    self.current_environ.name == 'PrgEnv-pgi'):
         #    self.omp_versions['PrgEnv-pgi'] = '200805'
-
+        self.maintainers = ['JG', 'MK']
+        self.tags = {'production'}
+        if self.current_system.name in {'dom', 'daint'}:
+            # get general info about the environment:
+            self.post_run = ['module list -t']
         self.perf_patterns = {
             'elapsed_time': sn.extractsingle(r'Elapsed Time\s*:\s+(\S+)',
                                              self.stdout, 1, float)
         }
-        self.reference_prgenv = {
-            'PrgEnv-gnu': (0.90, -0.6, None),
-            'PrgEnv-cray': (0.90, -0.6, None),
-            'PrgEnv-intel': (0.90, -0.6, None),
-            'PrgEnv-pgi': (18.0, -0.6, None),
-        }
         self.reference = {
             '*': {
-                'elapsed_time': (0, None, None)
+                'elapsed_time': (0, None, None, 'seconds')
             }
         }
-        self.post_run = ['module list -t']
-
-        self.maintainers = ['JG', 'MK']
-        self.tags = {'production'}
+        if lang == 'C++':
+            self.reference_prgenv = {
+                'PrgEnv-cray': (0.38, -0.6, None, 'seconds'),
```

Comment:
Use SI units, so `s` is better.

Reply:
https://github.com/eth-cscs/reframe/pull/819/commits/7924290da6343608a367f6192468344aa004528b